warehouse/query-core/src/main/java/datawave/query/transformer/UniqueTransform.java

```diff
@@ -66,66 +80,39 @@ public class UniqueTransform extends DocumentTransform.DefaultDocumentTransform
-     * be requested for date/time fields (@see UniqueFields). The logic will be used to get a query model to include the reverse mappings in the unique field
-     * set
+     * Update the configuration of this transform. If the configuration is actually changing, then the bloom filter will be reset as well.
      *
-     * @param logic
-     *            The query logic from whih to pull the query model
      * @param uniqueFields
-     *            The unique fields
-     * @param queryExecutionForPageTimeout
-     *            If this timeout is passed before since the last result was returned, then an "intermediate" result is returned denoting we are still looking
-     *            for the next unique result.
+     *            The new set of unique fields.
      */
-    public UniqueTransform(BaseQueryLogic<Entry<Key,Value>> logic, UniqueFields uniqueFields, long queryExecutionForPageTimeout) {
-        this(uniqueFields, queryExecutionForPageTimeout);
-        QueryModel model = ((ShardQueryLogic) logic).getQueryModel();
-        if (model != null) {
-            modelMapping = HashMultimap.create();
-            // reverse the reverse query mapping which will give us a mapping from the final field name to the original field name(s)
-            for (Map.Entry<String,String> entry : model.getReverseQueryMapping().entrySet()) {
-                modelMapping.put(entry.getValue(), entry.getKey());
-            }
-        }
-    }
-
-    public void updateConfig(UniqueFields uniqueFields, QueryModel model) {
-        if (this.uniqueFields != uniqueFields) {
-            uniqueFields.deconstructIdentifierFields();
-            if (!this.uniqueFields.equals(uniqueFields)) {
-                this.uniqueFields = uniqueFields;
-                log.info("Resetting unique fields on the unique transform");
-                this.bloom = BloomFilter.create(new ByteFunnel(), 500000, 1e-15);
-                if (log.isTraceEnabled()) {
-                    log.trace("unique fields: " + this.uniqueFields.getFields());
-                }
-            }
-        }
-        if (model != null) {
-            modelMapping = HashMultimap.create();
-            // reverse the reverse query mapping which will give us a mapping from the final field name to the original field name(s)
-            for (Map.Entry<String,String> entry : model.getReverseQueryMapping().entrySet()) {
-                modelMapping.put(entry.getValue(), entry.getKey());
+    public void updateConfig(UniqueFields uniqueFields) {
+        // only reset the bloom filter if changing the field set
+        if (!this.uniqueFields.equals(uniqueFields)) {
+            this.uniqueFields = uniqueFields.clone();
+            log.info("Resetting unique fields on the unique transform");
+            this.bloom = BloomFilter.create(new ByteFunnel(), 500000, 1e-15);
```

Comment:
since the bloom is only used when not using isMostRecent() can we avoid creating the bloom at all when that is set?

Reply:
probably, I will see if I can do that.

Reply:
done